huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go

```diff
@@ -1,32 +1,65 @@
-package huaweicloud
+package dds
 
 import (
 	"fmt"
 	"testing"
 
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
 
 	"github.com/chnsz/golangsdk/openstack/dds/v3/instances"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 )
 
+func getDdsResourceFunc(c *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := c.DdsV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("Error creating HuaweiCloud DDS client: %s ", err)
+	}
+
+	opts := instances.ListInstanceOpts{
+		Id: state.Primary.ID,
+	}
+	allPages, err := instances.List(client, &opts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	instances, err := instances.ExtractInstances(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if instances.TotalCount == 0 {
+		return nil, fmtp.Errorf("dds instance not found.")
+	}
+
+	insts := instances.Instances
+	found := insts[0]
```

Comment:
It is recommended to extract this part of the code as a function, the same code has been declared in `resource_huaweicloud_dds_instance_v3.go`.

Reply:
Although the code looks similar, the returned content is different. If we extract this as a function, only line25~32 can be used in the func. I don't think it is necessary.